core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/credentials/MapToAzureCredentialV4ResponseParameters.java

```diff
@@ -0,0 +1,38 @@
+package com.sequenceiq.cloudbreak.converter.v4.credentials;
+
+import static java.util.Objects.isNull;
+
+import java.util.Map;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.credentials.parameters.azure.AzureCredentialV4ResponseParameters;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.credentials.parameters.azure.RoleBasedResponse;
+import com.sequenceiq.cloudbreak.converter.AbstractConversionServiceAwareConverter;
+
+@Component
+public class MapToAzureCredentialV4ResponseParameters extends AbstractConversionServiceAwareConverter<Map<String, Object>,
+        AzureCredentialV4ResponseParameters> {
+
+    private static final String INVALID_MESSAGE_FORMAT = "Unable to set which based is the actual Azure credential parameter, since %s";
+
+    @Override
+    public AzureCredentialV4ResponseParameters convert(Map<String, Object> source) {
+        AzureCredentialV4ResponseParameters parameters = new AzureCredentialV4ResponseParameters();
+        parameters.setTenantId((String) source.get("tenantId"));
+        parameters.setSubscriptionId((String) source.get("subscriptionId"));
+        parameters.setAccessKey((String) source.get("accessKey"));
+        if (!isNull(source.get("appObjectId")) || !isNull(source.get("roleType"))) {
```

Comment:
This could be extracted to a method called `convertRoleBased`. It would still be a nice 9 line method, so not too short.